### PR TITLE
DDP-7724/DSM-show-all-questions-and-linter-fix

### PR DESCRIPTION
**Pull request by GiCharkviani**
https://broadinstitute.atlassian.net/browse/DDP-7724

### CODE REVIEW

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.model.ts [hunk 1]
```diff
@@ -24,4 +24,8 @@ export class ActivityData {
     }
     return null;
   }
+
+  checkAnswerStableId(stableId: string): QuestionAnswer {
```
Comment (gmakharat) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.model.ts]: `findAnswerByStableId`, or something similar would be more relevant name.

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildActivityData.service.ts [hunk 2]
```diff
@@ -14,10 +14,13 @@ export class buildActivityDataService {
...
 
     public buildActivity(activityDefinition: ActivityDefinition, activity: ActivityData): QuestionTypeModel[] {
-      for(const answer of activity.questionsAnswers) {
-        !([answer?.stableId, activityDefinition?.questions, activityDefinition.getQuestionDefinition(answer.stableId)]
-          .includes(null)) && this.checkQuestionType(answer, activityDefinition.getQuestionDefinition(answer.stableId));
-      }
+      activityDefinition?.questions.forEach(question => {
+        const questionAnswer = activity.checkAnswerStableId(question.stableId);
+
+        questionAnswer?.stableId && question ? this.checkQuestionType(questionAnswer, question)
```
Comment (gmakharat) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildActivityData.service.ts]: do we need `&& question`? I believe if question does not exist for each won't iterate.

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildActivityData.service.ts [hunk 3]
```diff
@@ -26,4 +29,12 @@ export class buildActivityDataService {
       activityQuestion && this.activityQuestions.push(activityQuestion.func({answer, question}));
     }
 
+    private noAnswer(question: QuestionDefinition): QuestionTypeModel {
```
Comment (gmakharat) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildActivityData.service.ts]: name does not show method's intent

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/models/data.model.ts [hunk 4]
```diff
@@ -52,10 +52,14 @@ export class Data {
     }
     return new Data(
       json.profile, json.status, json.statusTimestamp, json.dsm,
-      json.ddp, medicalProviders, json.activities, json.address, json.invitations, json.computed
+      json.ddp, medicalProviders, this.activities(json.activities), json.address, json.invitations, json.computed
     );
   }
 
+  private static activities(acts: ActivityData[]): ActivityData[] {
+    return acts.map(act => ActivityData.parse(act));
```
Comment (gmakharat) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/models/data.model.ts:60]: what's the point of having `ActivityData.parse(act)` if `acts` already contain `ActivityData` type objects?

**Review by ScalaChamp — APPROVED** [review]

**Pull request merged**